include/utils/PolyatomicDamageEnergyFunction.h

```diff
@@ -40,6 +37,19 @@ class PolyatomicDamageEnergyFunction : public PolyatomicDisplacementFunctionBase
    * by nu_i(E-T) - nu_i(E) \approx -T d(nu_i) / dE.
    */
   Real _taylor_series_threshold = 1e2;
+
+protected:
+  /// override the mapIndex function: flattens ijl to single index n
+  unsigned int mapIndex(unsigned int i, unsigned int /*j*/, unsigned int /*l*/) const override
+  {
+    return i;
```

Comment:
?

Reply:
I store damage functions as a flattened array indexed by `n`. If I compute damage energy functions, they depend only on a single isotope index `i` so in this case `n = i`, if you have damage functions they depend on `i` and `j` which are projectile and target isotope indices, and in case of derivatives of damage functions it depends additionally on `l` which is the isotope concentration that you differentiate with.

To move more stuff into the base class, I use the mapIndex function. In this case, the transformation from `i` to `n` is trivial.

Reply:
I think a better design would consist of multiple overloaded `mapIndex` members with 1, 2, and 3 parameters, where you have the overloads with the unsuitable parameter count throw an error.

Reply:
InverseMap is only used once to access a linear interpolation function. I could have a linear interpolation function like this:
```
linearInterpolationFromFlatIndex(energy, index)
```
Then I can remove inverse index function. `mapIndex` could be like this:
```
mapIndex(unsigned int i, unsigned int j = 0, unsigned int l = 0)
```
Then maybe throw a few asserts in mapIndex to ensure j=0 and l = 0.